google/cloud/spanner/timestamp_test.cc

```diff
@@ -495,7 +424,11 @@ TEST(Timestamp, ToChrono) {  // i.e., Timestamp::get<sys_time<Duration>>()
   auto const tp9 = kUnixEpoch - std::chrono::seconds(2123456789);
   EXPECT_EQ(tp9, ts_neg.get<sys_time<std::chrono::seconds>>().value());
 
-  auto const tp10 = kUnixEpoch - std::chrono::hours(2123456789 / 60 / 60);
+  // Timestamps drop precision by flooring toward negative infinity. Therefore,
+  // we need to craft our expectation by making sure our division also floors.
+  auto const floored_hours =
+      static_cast<std::chrono::hours::rep>(std::floor(-2123456789. / 60 / 60));
+  auto const tp10 = kUnixEpoch + std::chrono::hours(floored_hours);
```

Comment:
I would try to taking floating point (and `<cmath>`) out of the equation with something like:
```
auto const tp10 = kUnixEpoch - std::chrono::hours((2123456789 + 3600 - 1) / 3600);
```

Reply:
LGTM. Done.